### Coursera
#### Wesleyan University Data Analysis and Interpretation Specialization

Course 3: Regression Modeling in Practice<br>
Week 2: Test a Basic Linear Regression Model<br>
Author: Matt Clark

###### Introduction:

This week we're tasked to create a linear regression model. For our data set we'll continue to use the Outlook On Life survey. Our explanatory variable will be Level of Education (PPEDUCAT), a categorical variable. We'll collapse it into two levels. Our response variable will be Economic Optimism (W1_F6), as measured by how nearly a person expects to achieve the "American Dream" a categorical variable of ten levels. We will need to center this variable.

In [1]:
# import libraries

import pandas as pd
import numpy as np
import seaborn as sb
import scipy
import matplotlib.pyplot as plt
import statsmodels.api
import statsmodels.formula.api as smf
from pathlib import Path

In [2]:
# generate dataframe

root_dir = Path().resolve().parents[1]
df = pd.read_csv(str(root_dir)+'/mycodebook.csv', low_memory=False)

In [3]:
# bin the variable PPEDUCAT into two categories: High School or less, or something more than High School.

def collapse_variable (row):
   if row >= 3 :
       return 1
   else:
       return 0
    
df['PPEDUCAT'] = df['PPEDUCAT'].apply(collapse_variable)

In [4]:
# We generate a frequency table for this variable to check our coding.

check = df['PPEDUCAT'].value_counts(sort=False, dropna=False)
print (check)

0     919
1    1375
Name: PPEDUCAT, dtype: int64


In [5]:
# Linear Regression Model
print ("OLS regression model for the association between level of education and economic optimism")
reg1 = smf.ols('W1_F6 ~ PPEDUCAT', data=df).fit()
print (reg1.summary())

OLS regression model for the association between level of education and economic optimism
                            OLS Regression Results                            
Dep. Variable:                  W1_F6   R-squared:                       0.057
Model:                            OLS   Adj. R-squared:                  0.057
Method:                 Least Squares   F-statistic:                     139.3
Date:                Tue, 29 Sep 2020   Prob (F-statistic):           2.97e-31
Time:                        15:51:12   Log-Likelihood:                -5254.3
No. Observations:                2294   AIC:                         1.051e+04
Df Residuals:                    2292   BIC:                         1.052e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

###### Summary:

The results of our linear regression model indicate that a respondent's Level of Education ($\beta = 1.2027$, $p = 2.97 \times 10^{31}$) was not insignificantly associated with their Economic Optimism, with that association being positive.